File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/ArrayCombine.java
package com.onevgo.functions;

import cn.hutool.core.collection.IterUtil;

import java.util.Arrays;
import java.util.List;
import java.util.Map;

public class ArrayCombine {
    public static <K, V> Map<K, V> arrayCombine(Iterable<K> keys, Iterable<V> values) {
        return IterUtil.toMap(keys, values, true);
    }

    public static <K, V> Map<K, V> arrayCombine(K[] keys, V[] values) {
        return IterUtil.toMap(Arrays.asList(keys), Arrays.asList(values), true);
    }

    public static void main(String[] args) {
        List<String> a = Arrays.asList("green", "red", "yellow");
        List<String> b = Arrays.asList("avocado", "apple", "banana");
        Map<String, String> map = arrayCombine(a, b);
        System.out.println(map);
    }
}

File: 01_Language/01_Functions/php/array_combine.php
<?php

$a = array('green', 'red', 'yellow');
$b = array('avocado', 'apple', 'banana');
$c = array_combine($a, $b);

print_r($c);

$a = ['a1' => 'k1', 'a2' => 'k2', 'a3' => 'k3'];
$b = ['a2' => 'v2', 'a3' => 'v3', 'a1' => 'v1'];
$c = array_combine($a, $b);
print_r($c);
